src/generator.cs

```diff
@@ -7051,3 +7092,388 @@ string RenderType (Type t)
 	}
 	
 }
+
+public static class TypeManager {
```

Comment:
@spouliot @dalexsoto OK, I've moved the class to a new file. The downside is that there's a bunch of #ifdefs at the top of the file, which are now duplicated between generator.cs and generator-typemanager.cs, but in a later patch all those ifdefs will be automatically generated from the information available in frameworks.sources (eventually we might be able to use framework-specific ifdefs in monotouch-test as well, instead of platform-specific ifdefs).